src/Collection/Form/CollectionTypeExtension.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Symfony\UX\Collection\Form;
+
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\Extension\Core\Type\ButtonType;
+use Symfony\Component\Form\Extension\Core\Type\CollectionType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class CollectionTypeExtension extends AbstractTypeExtension
+{
+    public static function getExtendedTypes(): iterable
+    {
+        return [CollectionType::class];
+    }
+
+
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        /** @var FormInterface|null $prototype */
+        $prototype = $builder->getAttribute('prototype');
+
+        if (!$prototype) {
+            return;
+        }
+
+        // TODO add button only if `delete_type` is defined and set `delete_type` default to null?
+        if ($options['allow_delete']) {
+            // add delete button to prototype
+            // TODO add toolbar here to allow extension add other buttons
+            $prototype->add('deleteButton', $options['delete_type'], $options['delete_options']);
```

Comment:
when i use this code with a simple form type this throws: "You cannot add children to a simple form. Maybe you should set the option "compound" to true?"

```
$builder->add('fields', CollectionType::class, [
            'entry_type' => ChoiceType::class,
```

I'm not sure that it's a good idea to force a complex form just for frontend functionality. It seems more logical to me to put the delete button only in a template (like the maker bundle does the submit button https://github.com/symfony/maker-bundle/blob/main/src/Resources/skeleton/crud/templates/_form.tpl.php )

Reply:
Using not the button of the form theme make theming a lot harder. And I really recommend when possible use the button which is rendered by the form theme. So it works out of the box with the different form themes and don't require special form themes again for every css framework.